### PR TITLE
[PROT Warrior]Spell Reflect damage calc

### PR DESCRIPTION
**Pull request by abelito75**
Changed Spell Reflect to not be cast efficiency but rather did you use during magic damage.

### CODE REVIEW

File: src/parser/warrior/protection/CHANGELOG.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ import SpellLink from 'common/SpellLink';
 import { Abelito75 } from 'CONTRIBUTORS';
 import { change, date } from 'common/changelog';
 
-export default [  
+export default [   
+  change(date(2019, 10, 3),<>Spell reflect is now based off of damaged reduced. </>,Abelito75),
```
Comment (Juko8) [inline src/parser/warrior/protection/CHANGELOG.js]: Typo police 🚔 'damaged reduced' doesn't seem right

File: src/parser/warrior/protection/modules/spells/SpellReflect.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+import Analyzer from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import SCHOOLS from 'game/MAGIC_SCHOOLS';
+
+const debug = true;
```
Comment (Juko8) [inline src/parser/warrior/protection/modules/spells/SpellReflect.js]: Might want to have this as false in production

File: src/parser/warrior/protection/modules/spells/SpellReflect.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    abilityTracker: AbilityTracker,
+  };
+  
+  magicDamage = 0;
+  magicDamageReduced = 0;
+  totalDamage = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.lastCast = this.owner.fight.start_time/1000;
```
Comment (Juko8) [inline src/parser/warrior/protection/modules/spells/SpellReflect.js]: It doesn't seem like lastCast is being used for anything?

File: src/parser/warrior/protection/modules/spells/SpellReflect.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ 
+  suggestions(when) {
+    when(this.suggestionThresholds).addSuggestion((suggest, actual, recommended) => {
+      return suggest(
+        <>
+          Try to cast <SpellLink id={SPELLS.SPELL_REFLECTION.id} />  more often when magic damage is going out to take less damage.
+        </>
+      )
+        .icon(SPELLS.SPELL_REFLECTION.icon)
+        .actual(`${formatPercentage(actual)} % magic damage reduced`)
```
Comment (Juko8) [inline src/parser/warrior/protection/modules/spells/SpellReflect.js]: I think the phrasing here is a little unclear. Maybe 'You had Spell Reflect active during x% of magic damage taken" instead?

File: src/parser/warrior/protection/modules/spells/SpellReflect.js (hunk 2 above)
Comment (abelito75) [reply src/parser/warrior/protection/modules/spells/SpellReflect.js]: Oops missed that one

**Pull request merged**